MaterialLib/Component/Diffusion/createDiffusion.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file:   createDiffusion.cpp
+ */
+
+#include "createDiffusion.h"
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+#include "Diffusion.h"
+
+namespace MaterialLib
+{
+namespace Component
+{
+std::unique_ptr<Diffusion> createDiffusion(
+    BaseLib::ConfigTree const& config,
+    std::vector<std::unique_ptr<ProcessLib::ParameterBase>> const& parameters)
+{
+    //! \ogs_file_param{material__component__diffusion__type}
+    auto const type = config.getConfigParameter<std::string>("type");
+
+    if (type == "Constant")
+    {
+        auto const& constant_diffusion = ProcessLib::findParameter<double>(
+            config,
+            //! \ogs_file_param_special{material__component__diffusion__diffusion_parameter}
+            "diffusion_parameter", parameters, 1);
+
+        return std::make_unique<Diffusion>(constant_diffusion);
+    }
+
+    OGS_FATAL("The diffusion type %s is unavailable.\n",
+              "The available type is Constant.",
+              type.data());
```

Comment:
I think its better to use `type.c_str()` that gives you the non-modifiable standard C character array version of the string.

Reply:
Corrected.